core/include/scipp/core/groupby.h

```diff
@@ -16,19 +17,18 @@ namespace scipp::core {
 /// Stores the actual grouping details, independent of the container type.
 class SCIPP_CORE_EXPORT GroupByGrouping {
 public:
-  GroupByGrouping(Variable &&key, std::vector<std::vector<Slice>> &&groups)
+  using group = boost::container::small_vector<Slice, 4>;
```

Comment:
Specifically why now `small_vector`?

Reply:
I had hoped for performance improvements, but they were really minor. Since there is no reall downside I left it in (hoping that future improvements in other places would eventually make this more relevant).

When there are many small groups (say 1 million, which is a real application for grouping detectors into spectra) this avoid a lot of small allocations, provided that there are only few slices contributing to each group.